openedx/core/djangoapps/edx_global_analytics/tests/test_utils.py

```diff
@@ -0,0 +1,157 @@
+        )
+
+        self.assertEqual(result, 2)
+
+    def test_miss_statistics_query_fetch_instance_information_for_active_students_amount(self):
+        """
+        Verifies that fetch_instance_information raise `KeyError` if default statistics queries don't have
+        corresponding name (dict`s key).
+        """
+        activity_period = datetime.date(2017, 5, 15), datetime.date(2017, 5, 16)
+        cache_timeout = None
+
+        self.assertRaises(KeyError, lambda: fetch_instance_information(
+            'active_students_amount_day', 'active_students_amount_day', activity_period, cache_timeout
+        ))
+
+    def test_datetime_is_none_fetch_instance_information_for_active_students_amount(self):
+        """
+        Verifies that fetch_instance_information raise `TypeError` if needed datetime objects are missed
+        as activity period is None.
+        """
+        activity_period, cache_timeout = None, None
+
+        self.assertRaises(TypeError, lambda: fetch_instance_information(
+            'active_students_amount_day', 'active_students_amount', activity_period, cache_timeout
+        ))
+
+    def test_fetch_instance_information_for_students_per_country(self):
+        """
+        Verifies that students_per_country returns data as expected in particular period and accurate datetime.
+        """
+        last_login = timezone.make_aware(datetime.datetime(2017, 5, 15, 14, 23, 23), timezone.get_default_timezone())
+        countries = [u'US', u'CA']
+
+        for country in countries:
+            self.user = UserFactory.create(last_login=last_login)
+            self.profile = self.user.profile
+            self.profile.country = Country(country)
+            self.profile.save()
+
+        activity_period = datetime.date(2017, 5, 15), datetime.date(2017, 5, 16)
+        cache_timeout = None
+
+        result = fetch_instance_information(
+            'students_per_country', 'students_per_country', activity_period, cache_timeout)
+
+        self.assertItemsEqual(result, {u'US': 1, u'CA': 1})
+
+    def test_no_students_with_country_fetch_instance_information_for_students_per_country(self):
+        """
+        Verifies that students_per_country returns data as expected if no students with country.
+        """
+        last_login = timezone.make_aware(datetime.datetime(2017, 5, 15, 14, 23, 23), timezone.get_default_timezone())
+
+        for country in range(2):
+            self.user = UserFactory.create(last_login=last_login)
+
+        activity_period = datetime.date(2017, 5, 15), datetime.date(2017, 5, 16)
+        cache_timeout = None
+
+        result = fetch_instance_information(
+            'students_per_country', 'students_per_country', activity_period, cache_timeout)
+
+        self.assertEqual(result, {None: 0})
+
+
+class TestCacheInstanceData(TestCase):
+    """
+    Tests cover cache-functionality for queries results.
+    """
+    @staticmethod
+    def cache_instance_data_default_data():
```

Comment:
If this method is the same as in previous Test class -- extract it to the base class.
See https://refactoring.guru/extract-superclass